src/data-providers/dynamodb-marshaller.ts

```diff
@@ -0,0 +1,127 @@
+import { Pool, Schema } from '../types';
+import { POOL_SCHEMA, MAX_DYNAMODB_PRECISION } from '../constants';
+import { Marshaller, NumberValue} from '@aws/dynamodb-auto-marshaller';
+import BigNumber from 'bignumber.js';
+import { AttributeMap } from 'aws-sdk/clients/dynamodb';
+import util from 'util';
+
+/** Modify item to ensure it meets DynamoDB specifications */
+function sanitizeField(marshalledItem) {
+  // For numbers, trim to 38 decimals of precision
+  const sanitizedItem = {...marshalledItem};
+  if (sanitizedItem.N) {
+    sanitizedItem.N = new BigNumber(marshalledItem.N).precision(MAX_DYNAMODB_PRECISION).toString();
+  }
+
+  return sanitizedItem;
+}
+
+function marshallItem(schema: Schema, item) {
+  const marshalledItem = {};
+  Object.entries(schema).forEach(([key, keySchema]) => {
+    if (item[key]) {
+      switch (keySchema.type) {
+        case 'BigNumber':
+          marshalledItem[key] = {'N': item[key]};
+          break;
+        case 'Boolean':
+          marshalledItem[key] = {'BOOL': item[key]};
+          break;
+        case 'Number':
+          marshalledItem[key] = {'N': item[key]};
+          break;
+        case 'String':
+          marshalledItem[key] = {'S': item[key]}
+      }
+    }
+  });
+  return marshalledItem;
+}
+
+/** 
+ *  Does the final touches to convert a DynamoDB object back into a regular object
+ *  Converts NumberValue fields into regular strings/number
+ *  - Items specified as BigDecimal are converted to strings
+ *  - All others are converted to numbers
+ */
+function finalizeUnmarshalledItem(schema: Schema, item) {
+  const unmarshalledItem = {};
+  if (Array.isArray(item)) {
+    return item.map((value) => finalizeUnmarshalledItem({}, value));
+  }
+  if (!item || typeof item !== "object") {
+    return item;
+  }
+  Object.entries(item).forEach(([key, value]) => {
+    if (value instanceof NumberValue) {
+      if (schema[key]) {
+        switch (schema[key].type) {
+          case 'BigNumber':
+            unmarshalledItem[key] = value.value
+          break;
+          case 'Number':
+          default:
+            unmarshalledItem[key] = Number(value.value)
+          break;
```

Comment:
Should the Number case be handled here and why is there his break statement at the bottom?

Reply:
Number just falls through to the default. Don't need the break at the bottom I guess